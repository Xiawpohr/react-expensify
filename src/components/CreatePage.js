import React from 'react'
import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'
import ExpenseForm from './ExpenseForm.js'
import { addExpense } from '../actions/expenses.js'

export const CreatePage = (props) => {
  const onSubmit = (expense) => {
    props.addExpense(expense)
    props.history.push('/')
  }

  return (
    <div>
      <h2>Add Expense</h2>
      <ExpenseForm onSubmit={onSubmit} />
    </div>
  )
}

const mapDispatchToProps = (dispatch, props) => ({
  addExpense: (expense) => dispatch(addExpense(expense))
})

export default withRouter(connect(undefined, mapDispatchToProps)(CreatePage))
