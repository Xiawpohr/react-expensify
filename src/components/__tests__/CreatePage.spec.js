import React from 'react'
import { shallow } from 'enzyme'
import { CreatePage } from '../CreatePage.js'
import moment from '../__mocks__/moment.js'

let theme, addExpense, history, wrapper

beforeEach(() => {
  theme = {
    colors: { offWhite: '#f7f7f7' }
  }
  addExpense = jest.fn()
  history = { push: jest.fn() }
  wrapper = shallow(<CreatePage theme={theme} addExpense={addExpense} history={history} />)
})

test('should render Create Page correctly', () => {
  expect(wrapper).toMatchSnapshot()
})

test('should handle submit event', () => {
  const expense = {
    description: 'bill',
    amount: 1000,
    createdAt: moment(0).valueOf(),
    note: ''
  }
  wrapper.find('ExpenseForm').prop('onSubmit')(expense)
  expect(addExpense).toHaveBeenLastCalledWith(expense)
  expect(history.push).toHaveBeenLastCalledWith('/')
})
