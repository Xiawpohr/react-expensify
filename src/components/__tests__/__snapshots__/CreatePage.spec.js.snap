// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Create Page correctly 1`] = `
<div>
  <BackgroundColor
    color="#f7f7f7"
  >
    <Container
      mb={3}
      py={3}
    >
      <h1>
        Add Expense
      </h1>
    </Container>
  </BackgroundColor>
  <Container>
    <ExpenseForm
      onSubmit={[Function]}
    />
  </Container>
</div>
`;
